src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.KeyTrans.cs

```diff
@@ -142,20 +140,42 @@ internal byte[] DecryptCek(X509Certificate2 cert, RSA privateKey, out Exception
                     recipient.RecipientIdentifierType.ToString());
             }
 
-            RSAEncryptionPadding padding;
+            RSAEncryptionPadding padding = recipient.RSAEncryptionPadding ?? recipient.Certificate.GetKeyAlgorithm() switch {
```

Comment:
I did some testing with RSAES-OAEP certificates and it's not that well supported, but I preserved the existing behavior. I made an RSAES_OAEP certificate and found that:

1. The Windows PAL will not handling them correctly for `EnvelopedCms`. A switch is not correctly handling that case but it seems fixable.
2. macOS and openSSL don't like these kinds of certificates at all. openssl fails with envelope `routines:pkey_set_type:unsupported algorithm` when opening the certificate.